### ISSUE TITLE
Predicates in a "hasLabel" traversal are broken

### ISSUE THREAD

**Issue by Sushisource** [lambdazen/bitsy#11]
I'd set up a traversal like so:

```java
final Set<String> names = implementors.stream().map(GraphQLObjectType::getName)
                                               .collect(Collectors.toSet());
traversal = graph.traversal()
                 .V()
                 .hasLabel(P.within(names));
```

Attempting to run this traversal fails with 

```
java.util.HashSet cannot be cast to java.lang.String
```

In `com.lambdazen.bitsy.gremlin.BitsyGraphStep.lambda$lookupVertices$9(BitsyGraphStep.java:67)`

This is version 3.0.2

**Comment by Sushisource** [comment]
For reference the same code works fine with http://tinkerpop.apache.org/docs/current/reference/#tinkergraph-gremlin

**Comment by lambdazen** [comment]
This has to do with the Bitsy-Gremlin integration which is a little harder to to in TP3 than TP2. 

I have a feeling that line 165 in
https://github.com/apache/tinkerpop/blob/453665d34e9a990def47e1ed35d18013b3286cbf/neo4j-gremlin/src/main/java/org/apache/tinkerpop/gremlin/neo4j/structure/trait/NoMultiNoMetaNeo4jTrait.java

needs to go after line 66 & 102 in 
https://github.com/lambdazen/bitsy/blob/61ebcdb1f5f62dfe2ee301cccef19e2ff8d07fb4/src/main/java/com/lambdazen/bitsy/gremlin/BitsyGraphStep.java

Although, I'm not clear why this line was skipped in the first place... 

Would you be able to send me a quick standalone test file for this? I can convert it to a test case and make sure that all tests pass after the change. 

Regards,

Sridhar.

**Comment by lambdazen** [comment]
Thanks for the test. 

I have prepared 3.0.3 snapshot, with the fix here: https://oss.sonatype.org/content/repositories/snapshots/com/lambdazen/bitsy/bitsy/3.0.3-SNAPSHOT/

Hope this works for you. I'll release this version later this week. 

Regards,

Sridhar.

**Comment by lambdazen** [comment]
I have released version 3.0.3. https://oss.sonatype.org/content/groups/public/com/lambdazen/bitsy/bitsy/3.0.3/

Please reopen this issue if it still persists. 

Regards,

Sridhar.
